### PR TITLE
Labels pyramid

### LINKED ISSUE #538: Help with starfish.display for large images & mask collections
## 🐛 Bug

I would like to load the segmented Allen data in napari. These data consist of ~2200 regions of interest over a field of view of shape (27084, 20825)

When I attempt to use napari to load these data, the images are too large and the mask collections throw errors. You can view my attempts [here](https://github.com/spacetx/starfish/blob/2d61bdd0490f22db2837aa52249baaf66d40f18b/notebooks/manual_segmentation_example.ipynb), beginning under the "GeoJSON to SegmentationMaskCollection" heading. Please forgive the messy notebook. 

## To Reproduce

unzip the file downloaded from s3: `s3://starfish.data.spacetx/smFISH/mouse/contributor_results/napari_large_image.zip`
```
70167526 Sep 11 16:40 mFISH__001_registered-downsampled.png
3390209  Sep 16 14:56 mFISH_labels.png
```
To reproduce, load the label image and attempt to show the resulting data in napari. 

```
import skimage.io
labels = skimage.io.imread('mFISH_labels.png')
```
## Expected behavior

I'd like to be able to view my images & masks without size constraints. 

## Environment

 - napari Version (e.g., 1.0): 0.1.3 and 0.1.5 both throw the errors
 - OS (e.g., Linux): OS X 10.14
 - Python version: 3.7

### PR DESCRIPTION
**Pull request by sofroniewn**
# Description
This PR extends pyramid support to the Labels layer following the same conventions as the Image. It does this by making Labels subclass Image which actually reduces a lot of duplicate code.

I also make the thumbnails now always be blended into a black background.

It also introduces a new concept of layers being "editable" or not and sets the Labels layer to not being editable when it is a pyramid or being rendered in 3D (which would have caused bugs). One day we may extend painting to the pyramid or 3D but for now those are more advanced features. People have also requested the ability to disable editing for datasets where it doesn't make sense. When disabled the paint / fill / pick buttons now have a different background color to indicate they are no longer selectable. If people are ok with this I will extend the editable property to the shapes and points layers where we also need to disable editing for 3D rendering.

Here is what this looks like for the 3D rendering case:

editable_labels

This closes #538. @ambrosejcarr does this look good to you? 

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How has this been tested?

- [x] added pyramid test to the Labels tests

## Final checklist:
- [x] My PR is the minimum possible work for the desired functionality
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by sofroniewn** [comment]
I'v gone ahead and extended "editable" mode to Points and Shapes within this PR.

File: napari/layers/image/tests/test_pyramid.py
Comment (jni) [inline napari/layers/image/tests/test_pyramid.py]: Why this change?

**Review by jni — APPROVED** [review]
Brilliant! I love how much the code is simplified by this change! :tada:

My only concern is that the buttons styling to me suggests "pressed" rather than "disabled". Typically this is shown by buttons having lower contrast, but currently that would make it confusing compared to the styling for "selected". We can leave button styling details for a later PR though.

File: napari/layers/image/tests/test_pyramid.py
Comment (sofroniewn) [reply napari/layers/image/tests/test_pyramid.py]: I added the ability of the layer to "trim" the pyramid so that if you pass it say a pyramid that keeps going until the final layer is `1x1` the smallest resolution it would actually show is `63x63` (which is also the resolution it would use for the thumbnail). If all the layers are very small then the pyramid just takes the bottom two, which it is doing here. I've added a comment inside the test.

**Comment by sofroniewn** [comment]
fair enough on the stylings, but let's wait till a second review with @liaprins-czi to improve

**Comment by royerloic** [comment]
Very cool @sofroniewn !

**Pull request merged**
